src/lib/y2packager/medium_type.rb

```diff
@@ -33,6 +33,11 @@ def type
         @type ||= detect_medium_type
       end
 
+      # Allows to overwrite detected medium type. Useful e.g. when upgrade of
+      # registered system with Full medium should act like Online medium.
+      # @return [Symbol] possible values are `:online`, `:offline` and `:standard`
+      attr_writer :type
```

Comment:
Um, overriding the detected medium type is a quite dangerous operation, we should not do it silently.
I'd prefer some logging here.

And we should probably also validate the input value, to make sure it is one of the allowed symbols.